### DISCUSSION TITLE
Is it possible to generate a file per type instead of one mega-file with everything?

### DISCUSSION THREAD

**Discussion by JaffParker** [dotansimha/graphql-code-generator#5077] [Ideas]
Hi. Been using and loving this. But like many others, I'm concerned with the fact that my app now has to drag a massive file with it on every page load, it's not ideal. I tried the near-operation-file-preset, but it still generates a mega file and references it from each operation.

The devs mentioned that plugin authors now can generate multiple files, but after going through the list of plugins, I didn't find anything that would do this yet. What I have in mind is basically creating a file for each object type, input type and enum, then one for scalars. That way each operation would only import the code relevant to it, nothing extra.

I haven't looked at plugin authoring yet, whether I'd be able or not to create one. If someone has, what do you think? Can you think of any drawbacks with such approach?

**Comment by geekox86** [comment]
Thank you @JaffParker. This is very much my question also.

May I add also: is there a way to transform individual GraphQL schema files to some other format without actually having to specify them one by one in the codegen config file as outputs? The problem is that a custom plugin can only return one string.

**Comment by geekox86** [comment]
Just thought this [discussion](https://github.com/dotansimha/graphql-code-generator/discussions/4922) might also be relevant.

**Reply by JaffParker** [reply]
That's actually good. I didn't realize it's possible to create a preset,  but of course! There's just no docs for it

**Reply by dotansimha** [reply]
We do have docs for presets: https://graphql-code-generator.com/docs/presets/near-operation-file 
We currently don't have a preset for generating per each schema typed. Might be tricky because of circular dependencies. PRs are welcome!
